src/getData.ts

```diff
@@ -0,0 +1,69 @@
+
+import { Children } from 'react';
+import ApolloClient from 'apollo-client';
+import flatten = require('lodash.flatten');
+import assign = require('object-assign');
+
+export default function getData(tree, props: any = {}): Promise<any> {
+  let client;
+  let store;
+  function getQueriesFromTree(components, queries = []) {
+    Children.forEach(components, (child: any) => {
+      if (!child) {
+        return;
+      }
+
+      if (!store && child.props && child.props.store) {
+        store = store;
+      }
+
+      // find the client in the tree
+      if (!client && child.props && child.props.client instanceof ApolloClient) {
+        client = child.props.client as ApolloClient;
+        if (!store) {
+          client.initStore();
+          store = client.store;
+        }
+      }
+
+      if (child.type && typeof child.type.mapQueriesToProps === 'function') {
+        queries.push(child.type.mapQueriesToProps);
+      }
+      if (child.props && child.props.children) {
+        getQueriesFromTree(child.props.children, queries);
+      }
+    });
+    return queries;
+  }
+
+  const rawQueries = getQueriesFromTree(tree);
```

Comment:
I think exporting a few options to allow you to build your own rendering process, or just use react-apollo would be best:

i.e:
1. `getQueries, runQueries` for a fully custom solution
2. `getData` which runs all queries tagged as SSR
3. a `renderWithData` that returns back the full SSR payload with the initial state injected into the markup.